client.go

```diff
@@ -357,3 +367,153 @@ func prepareURL(url *url.URL, path string) (*url.URL, error) {
 
 	return u, nil
 }
+
+// prepareQueryParameters Allows for easy preparation of query string
+func (cl *Client) prepareQueryParameters(params interface{}) (queryString string, err error) {
+	v, err := query.Values(params)
+
+	// If we do have account switch key - we will add it and toggle ASK back to disabled
+	if cl.accountSwitchEnabled == true {
+		v.Add("accountSwitchKey", cl.accountSwitchKey)
+		cl.accountSwitchEnabled = false
+	}
+
+	if err != nil {
+		return "", err
+	}
+
+	return v.Encode(), nil
+}
+
+// makeAPIRequest creates an HTTP request that can be sent to Akamai APIs. It will handle security headers and signinig of the request.
+//
+func (cl *Client) makeAPIRequest(method, path string, queryParams, structResponse, structRequest interface{}, headers map[string]string) (*ClientResponse, error) {
+
+	log.Debug("[NewRequest]::Prepare URL for http request")
+	targetURL, _ := prepareURL(cl.baseURL, path)
+
+	log.Debug("[NewRequest]::Create http request")
+	req, err := http.NewRequest(method, targetURL.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	/*
+		Modify request for POST/PUT
+	*/
+	if method == http.MethodPost || method == http.MethodPut {
+		log.Info("Prepare request body object")
+		log.Debug("[NewRequest]::Method is POST/PUT")
+		log.Debug("[NewRequest]::Marshal request object")
+
+		reqType := reflect.TypeOf(structRequest)
+		log.Debug(fmt.Sprintf("[NewRequest]::Object request provided type ( %s ) ", reqType))
+
+		bodyBytes, err := json.Marshal(structRequest)
+		if err != nil {
+			return nil, err
+		}
+		bodyReader := bytes.NewReader(bodyBytes)
+
+		log.Debug("[NewRequest]::Body object added to request")
+		req.Body = ioutil.NopCloser(bodyReader)
+		req.ContentLength = int64(bodyReader.Len())
+
+		log.Debug("[NewRequest]::Body object is:" + string(bodyBytes))
+		log.Debug("[NewRequest]::Set header Content-Type to 'application/json' ")
+		req.Header.Set("Content-Type", "application/json")
+
+	}
+
+	/*
+		Add headers
+	*/
+	if headers != nil {
+		log.Debug("[NewRequest]::Add extra headers")
+		for k, v := range headers {
+			log.Debug(fmt.Sprintf("[NewRequest]::Adding %s:%s", k, v))
+			req.Header.Add(k, v)
+		}
+	}
+
+	/*
+		Add query params
+	*/
+	if queryParams != nil {
+		log.Debug("[NewRequest]::Add query string parameters")
+		rawQueryString, queryStringError := cl.prepareQueryParameters(queryParams)
+
+		req.URL.RawQuery = rawQueryString
+
+		if queryStringError != nil {
+			log.Debug("[NewRequest]::Error adding query string parameters")
+			log.Debug(fmt.Sprintf("[NewRequest]:: %s", queryStringError.Error()))
+			return nil, queryStringError
+		}
+	}
+
+	/*
+		Add signature header
```

Comment:
Why not to move signature header closer to previous header block to have headers in same place and near

Reply:
Fixed